pkg/datapath/iptables/iptables.go

```diff
@@ -850,6 +852,33 @@ func (m *IptablesManager) InstallProxyRules(proxyPort uint16, ingress bool, name
 	m.Lock()
 	defer m.Unlock()
 
+	backoff := backoff.Exponential{
+		Min:  20 * time.Second,
+		Max:  3 * time.Minute,
```

Comment:
Correct, in the worst case no one else will be able to update the ruleset for the 3 minutes window (and the caller will block for that time). At the same time I'm not sure if making it async is a good strategy, as to my understanding the caller will assume that once this function returns the rules are in place (unless there are errors)

Reply:
What if we moved the `m.Lock()` & `m.Unlock()` around `m.doInstallProxyRules()` inside the loop? Would there be any downside to that? It would avoid holding a lock while we're sleeping.

Reply:
Good shout :+1: I can't see any downside of that as we are not accessing any `IptablesManager` state in `InstallRules`/`InstallProxyRules` so I moved the locks inside `doInstallRules`/`doInstallProxyRules`